src/application_config/configuring_base.py

```diff
@@ -21,49 +22,45 @@
 
 
 @dataclass(frozen=True)
-class ConfigSectionBase(ABC):
-    """Base class for all ConfigSection classes"""
+class ConfigSectionBase:
+    """Base class for all ConfigSection classes (so that we can bound a TypeVar)"""
 
 
-# The next line has a type: ignore because MyPy has a problem with @abstractmethod
-# If you comment the line with @abstractmethod, you can do type checking
 @dataclass(frozen=True)
-class ConfigBase(ABC):
+class ConfigBase:
     """Base class for main Config class"""
 
-    @staticmethod
-    @abstractmethod
-    def get_app_basename() -> str:
-        """Return the application base name.
-
-        This name, with a preceding dot, will also be the folder name in the home directory
-        that will store the config.
-        """
+    @classmethod
+    def default_config_foldername(cls: type[TConfig]) -> str:
+        """Return the class name without 'Config' and with a preceding dot as default for the folder that will hold the config file."""
+        return f".{cls.__name__.replace('Config', '')}"
 
     @classmethod
-    def get_configfile_path(cls: type[TConfig]) -> Path:
-        """Return the fully qualified path for the configfile: e.g. ~/.stockwatch/stockwatch-config.toml"""
-        return Path.home() / f".{cls.get_app_basename()}" / "config.toml"
+    def defaut_config_filepath(cls: type[TConfig]) -> Path:
+        """Return the fully qualified path for the configfile: e.g. ~/.Example/config.toml"""
+        return Path.home() / f"{cls.default_config_foldername()}" / "config.toml"
```

Comment:
I don't think we should do `Path.home()` here, we might want some fully qualified path (`/opt/app/config.toml`). So the `Path.home()` should probably move to `default_config_foldername`)

Reply:
I don't understand what you mean. Clients can override this method with a fully qualified name (like what you suggested).

Reply:
Clients can override this method and return `None` if they want to enforce the use of an explicit config file path in `get`